examples/tutorials/date_time_charts.py

```diff
@@ -0,0 +1,366 @@
+"""
+Plotting datetime charts
+========================
+
+Creating datetime charts is handled by :meth:`pygmt.Figure.basemap`.
+
+Plotting datetime data points is handled by :meth:`pygmt.Figure.basemap`.
+"""
+# sphinx_gallery_thumbnail_number = 0
+
+import datetime
+
+import numpy as np
+import pandas as pd
+import pygmt
+import xarray as xr
+
+########################################################################################
+# Datetime Input Types
+# --------------------
+#
+# PyGMT accepts a variety of datetime objects to plot data and create charts.
```

Comment:
The TOC of the tutorial is now:

- Datetime
- ISO
- Mixing Datetime and ISO
- pandas.date_range
- xarray.DataArray
- numpy.datetime64
- Generating Region
- Setting Primary and Secondary Time Axes

In this paragraph, the order is `datatime`, `pandas`, `numpy`, `xarray`, `ISO`. How about having the same order?